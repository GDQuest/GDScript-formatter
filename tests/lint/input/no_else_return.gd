#region Good

func good(value):
	if value < 0:
		print("Negative value")
		return -1
	
	if value > 100:
		print("Too large")
		return 100
	
	return value

#endregion

#region Bad

func bad(value):
	if value < 0:
		print("Negative value")
		return -1
	elif value > 100:
		print("Too large")
		return 100
	else:
		return value

#endregion