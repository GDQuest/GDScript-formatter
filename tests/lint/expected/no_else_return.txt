tests/lint/input/no_else_return.gd:22:no-else-return:warning: Unnecessary 'elif' after 'if' block that ends with 'return'. Use 'if' instead
tests/lint/input/no_else_return.gd:25:no-else-return:warning: Unnecessary 'else' after 'if'/'elif' blocks that end with 'return'